### PR TITLE
2020 10 16

### PR DESCRIPTION
**Pull request by Anishluke92**
… proper algorithm to write efficient code!

### CODE REVIEW

File: SortByLength.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Now we got to know all the word size in that sentence. Lets rearrange the sentence by sorting the words by size in ascending order. example,
+"my dear hello world"
+=end
+def sort_by_length(strings)
+
+  string = strings.split(" ")
+  string.sort_by(&:length).join(" ")
+
+ 
+end
```
Comment (danielpaul) [inline SortByLength.rb]:
```suggestion
def sort_by_length(string)
  string.split(" ").sort_by(&:length).join(" ")
end
```

File: SharingIsCaring.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+show_the_love([4, 1, 4]) ➞ [3, 3, 3]
+
+show_the_love([16, 10, 8]) ➞ [12, 7.5, 14.5]
+
+show_the_love([2, 100]) ➞ [27, 75]
+Notes
+There will only be one smallest number in a given array.
+=end
+def love(removed, numbers)
+  numbers.map{ |number|
```
Comment (danielpaul) [inline SharingIsCaring.rb]:
```suggestion
  numbers.map do |number|
  
  end
```

use this syntax for multi line blocks. Does the same thing but its better ruby syntax than the {}

File: CountTheLoneOnes.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+then we check the next number 1 with the previous number 0. we found another lone 1 so we increase the count +1.
+so in total the count would be 2.
+5 Lets take the next number 1191, First lets take first number 1 compare it with next number 1 and we found repetition, then we compare the next number 9 with the previous number 1 and we found a different number(non repeating number).
+6.To the last we compare the next number 1 with the previous number 9, then we found our first lone 1. we should then increase the count +1.
+in total the count would be 1
+=end
+def lone(integer, len)
+  count = 0
+  integer.each_with_index do |number, index|
+    if index == 0 && number == "1" && integer[index+1] != '1'  || index == len  && number == "1" && integer[index-1] != '1'
```
Comment (danielpaul) [inline CountTheLoneOnes.rb]: if you converted the input to an integer, then you don't have to work with strings here. you can just check if number == 1 etc.

File: CountTheLoneOnes.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  end
+  count
+end
+
+def count_lone_ones(integer)
+  integer= integer.to_s.split("")
+  return 0  if !integer.include?("1")
+  len = (integer.length)-1
+
+  lone(integer, len)
```
Comment (danielpaul) [inline CountTheLoneOnes.rb]: we don't need a new method for this? methods are only useful to repeat a particular action and call it multiple times. if you are only going to call it once, then creating two methods would actually be less efficient. we then have part one and part two of the same algorithm then.

**Review by danielpaul — CHANGES_REQUESTED** [review]

File: CountTheLoneOnes.rb (hunk 4 above)
Comment (Anishluke92) [reply CountTheLoneOnes.rb]: Got it

File: SharingIsCaring.rb (hunk 2 above)
Comment (Anishluke92) [reply SharingIsCaring.rb]: Okay.

**Review by danielpaul — APPROVED** [review]

**Pull request merged**